New sample: Use NAA SSO from event-based code in Outlook add-in (#832)
OfficeDev/Office-Add-in-samples: Samples/auth/Outlook-Event-SSO-NAA/README.md

extensions:
  contentType: samples
  technologies:
    - Add-ins
  createdDate: "08/20/2024 10:00:00 AM"
description: "This sample shows how to implement SSO in an event in an Outlook add-in by using nested app authentication."
---

# Use SSO in events in an Outlook add-in using nested app authentication (preview)

## Summary

This sample shows how to implement SSO in an event in an Outlook add-in. It uses the Microsoft Authentication Library for JavaScript (MSAL.js) and nested app authentication (NAA) to access Microsoft Graph APIs for the signed in user. The sample displays the signed in user's name as a signature in the body of a new email or calendar item.

> [!IMPORTANT]
> Nested app authentication is currently in preview. To try this feature, you need to join the [Microsoft 365 Insider Program](https://insider.microsoft365.com/join) and choose **Beta Channel**. Don't use NAA in production add-ins. We invite you to try out NAA in test or development environments and welcome feedback on your experience through GitHub (see https://github.com/OfficeDev/office-js/issues).

## Features

- Use MSAL.js NAA to get an access token for the signed in user to call Microsoft Graph APIs.

> - Get an access token through NAA in the OnNewMessageCompose and OnNewAppointmentOrganizer events.

Reviewer: Not that I think these terms will be autotranslated, but we should wrap event names in code ticks.
```suggestion
- Get an access token through NAA in the `OnNewMessageCompose` and `OnNewAppointmentOrganizer` events.
```

> - Add a signature to an email or calendar invite with the signed in user's name.

Reviewer: Linda should correct me if I'm wrong, but I believe when signed-in is an adjective, it should be hyphenated.
```suggestion
- Add a signature to an email or calendar invite with the signed-in user's name.
```

## Applies to

> - Outlook (Beta Channel for classic Outlook only, new Outlook coming soon)

Reviewer: So Windows only?

For more information on supported platforms, see [NAA supported accounts and hosts](https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in#naa-supported-accounts-and-hosts).

## Prerequisites

- Office connected to a Microsoft 365 subscription (including Office on the web).

> - You need to join the [Microsoft 365 Insider Program](https://insider.microsoft365.com/join) to use the NAA preview features. Choose the **Current Channel (Preview)** insider level.

Reviewer: The earlier instruction in the Important note says "Beta Channel". Should this match?

      ![The redirect URIs link.](./assets/ui-add-redirect-link.png)

1. In the **Single-page application Redirect URIs** section, select **Add URI**.

      ![The Add URI link.](./assets/ui-add-redirects-link.png)

1. Enter the new URI value `https://localhost:3000/taskpane.html` and select **Save**.

      ![The completed redirects in the application registration.](./assets/ui-completed-redirects.png)

For more information on how to register your application, see [Register an application with the Microsoft Identity Platform](https://learn.microsoft.com/graph/auth-register-app-v2).

### Configure the sample

1. Clone or download this repository.
1. From the command line, or a terminal window, go to the root folder of this sample at `/samples/auth/Outlook-Event-SSO-NAA`.
1. Open the `src/launchevent/launchevent.js` file.
1. Replace the placeholder "Enter_the_Application_Id_Here" with the Application ID that you copied.
1. Open the `src/taskpane/taskpane.js` file.
1. Replace the placeholder "Enter_the_Application_Id_Here" with the Application ID that you copied.
1. Save the file.

## Run the sample

> First ensure you have signed in and consented to the add-in's scopes. Once you approve consent, you no longer need to do these steps.

Reviewer: Suggestion to clarify you're not talking about the other steps listed below.
```suggestion
First ensure you have signed in and consented to the add-in's scopes. Once you approve consent, you no longer need to do those steps.
```

1. Run the following commands.

    `npm install`
    `npm run build:dev`
    `npm run start`

    This will start the web server and sideload the add-in to Outlook.

> > [!IMPORTANT]
> > The dev build uses the `https://localhost:3000/public/launchevent.js` path. If you change to the production build, you also need to change the URL to `https://localhost:3000/launchevent.js` in the `manifest.xml` file.

Reviewer: I think this needs to be intended so it's under '1.'
```suggestion
    > [!IMPORTANT]
    > The dev build uses the `https://localhost:3000/public/launchevent.js` path. If you change to the production build, you also need to change the URL to `https://localhost:3000/launchevent.js` in the `manifest.xml` file.
```

1. Start Outlook (classic) and sign in.
1. Open an existing email item.
1. Choose "Show Task Pane" from the ribbon. This will open the task pane of the add-in.
1. Select the **Sign in** button to sign in. You may be prompted to consent to the scopes of the add-in. The task pane will indicate it signed in by displaying your user name.

Now you can use the event-based code.

1. Create a new email. The add-in will automatically add a signature with your signed in name.

> > [!Note]

Reviewer: ```suggestion
> [!NOTE]
```

---
page_type: sample
urlFragment: outlook-event-sso-naa
products:
  - office
  - office-outlook
languages:
  - javascript
extensions:
  contentType: samples
  technologies:
    - Add-ins
  createdDate: "08/20/2024 10:00:00 AM"
description: "This sample shows how to implement SSO in an event in an Outlook add-in by using nested app authentication."
---

> # Use SSO in events in an Outlook add-in using nested app authentication (preview)

Reviewer 2: ```suggestion
# Implement SSO in events in an Outlook add-in using nested app authentication (preview)
```
Just trying to avoid use and using in the same sentence. I suggest swapping the verb with the first sentence in the first paragraph. This won't affect SEO.

## Summary

> This sample shows how to implement single sign-on (SSO) in an event in an Outlook add-in. It uses the Microsoft Authentication Library for JavaScript (MSAL.js) and nested app authentication (NAA) to access Microsoft Graph APIs for the signed-in user. The sample displays the signed-in user's name as a signature in the body of a new email or calendar item.

Reviewer 2: ```suggestion
This sample demonstrates how to use single sign-on (SSO) in an Outlook add-in event using the Microsoft Authentication Library for JavaScript (MSAL.js) and nested app authentication (NAA) to access Microsoft Graph APIs for the authenticated user. The sample displays the signed-in user's name as a signature in the body of a new email or calendar item.
```

> [!IMPORTANT]

> > Nested app authentication is currently in preview. To try this feature, you need to join the [Microsoft 365 Insider Program](https://insider.microsoft365.com/join) and choose **Beta Channel**. Don't use NAA in production add-ins. We invite you to try out NAA in test or development environments and welcome feedback on your experience through GitHub (see https://github.com/OfficeDev/office-js/issues).

Reviewer 2: ```suggestion
> Nested app authentication is currently in preview. To try this feature, join the [Microsoft 365 Insider Program](https://insider.microsoft365.com/join) and choose **Beta Channel**. Don't use NAA in production add-ins. We invite you to try out NAA in test or development environments and welcome feedback on your experience through GitHub (see https://github.com/OfficeDev/office-js/issues).
```

      ![The redirect URIs link.](./assets/ui-add-redirect-link.png)

1. In the **Single-page application Redirect URIs** section, select **Add URI**.

      ![The Add URI link.](./assets/ui-add-redirects-link.png)

1. Enter the new URI value `https://localhost:3000/taskpane.html` and select **Save**.

      ![The completed redirects in the application registration.](./assets/ui-completed-redirects.png)

For more information on how to register your application, see [Register an application with the Microsoft Identity Platform](https://learn.microsoft.com/graph/auth-register-app-v2).

### Configure the sample

1. Clone or download this repository.
1. From the command line, or a terminal window, go to the root folder of this sample at `/samples/auth/Outlook-Event-SSO-NAA`.
1. Open the `src/launchevent/launchevent.js` file.
1. Replace the placeholder "Enter_the_Application_Id_Here" with the Application ID that you copied.
1. Open the `src/taskpane/taskpane.js` file.
1. Replace the placeholder "Enter_the_Application_Id_Here" with the Application ID that you copied.
1. Save the file.

## Run the sample

> First ensure you have signed in and consented to the add-in's scopes. Once you approve consent, you no longer need to do those steps.

Reviewer 2: ```suggestion
First ensure you're signed in and consent to the add-in's scopes. Once you approve consent, you no longer need to do those steps.
```
